### PR TITLE
componentsValues: componentValues for many components

### PR DESCRIPTION
**Pull request by vhf**
Idea for solving #37

### CODE REVIEW

File: src/datacube.ts [hunk 1]
```diff
@@ -213,6 +213,86 @@ export class DataCube {
...
+        generateLangCoalesce(labelBinding, this.languages),
+      ],
+      type: "query",
+    };
+
+    const generator = new SparqlGenerator({ allPrefixes: true });
+    const sparql = generator.stringify(query);
+    const rawResult: Array<{component: NamedNode, label: Literal, value: NamedNode}>
+      = await this.fetcher.select(sparql);
+    const resultMap = new Map();
```
Comment (herrstucki) [inline src/datacube.ts]: Would `WeakMap` be more appropriate than `Map` here? Or just an array that stores the values in the same order as the `Component[]` argument? Either way is fine I think …

File: src/datacube.ts [hunk 2]
```diff
@@ -213,6 +213,86 @@ export class DataCube {
...
+
+  /**
+   * Retrieve all the possible values a [[Component]] ([[Dimension]], [[Measure]], [[Attribute]]) can have.
+   * Similar to [[DataCube.componentsValues]] but for a single Component.
+   * See also [[Query.componentValues]] and the examples folder.
+   * ```js
+   * const values = await dataCube.componentValues(sizeClasses);
+   * ```
+   * @param {Component} component
+   * @returns {Promise<Array<{label: Literal, value: NamedNode}>>}
```
Comment (herrstucki) [inline src/datacube.ts]: Shouldn't that be `Promise<Array<{label: Literal, value: NamedNode | Literal}>>`? E.g. in the case of the years the values are literals.

File: src/datacube.ts [hunk 3]
```diff
@@ -213,6 +213,86 @@ export class DataCube {
    * @param {Component} component
    * @returns {Promise<Array<{label: Literal, value: NamedNode}>>}
    */
+  public async componentsValues(components: Component[]):
+    Promise<Map<Component, Array<{label: Literal, value: NamedNode}>>> {
```
Comment (herrstucki) [inline src/datacube.ts]: Same question about the type of `value` as below …

File: src/datacube.ts (hunk 1 above)
Comment (vhf) [reply src/datacube.ts]: I don't mind using `WeakMap`. Map/WeakMap seem a bit more useful than an array to me.

File: src/datacube.ts (hunk 2 above)
Comment (vhf) [reply src/datacube.ts]: Good catch!

**Pull request merged**